Added loading indicator when the app is connecting to the server

susonthapa opened the pull request:
**Changes**
I have added a loading indicator that will show when the app is connecting to the server so that users aren't left with a black screen. I have used `ContentLoadingProgressBar` which will try to ensure that the loading process is flicker-free. 

**Issues**
Relates #900 

**Demo**
When it takes 5 seconds to connect to the server the progress indicator is visible. 

https://user-images.githubusercontent.com/33973551/204138567-2178eaf3-2d7b-42de-b9d5-baf8e7c0d80d.mp4

When the connection is quick then the progress indicator won't be visible to reduce flicker.

https://user-images.githubusercontent.com/33973551/204138592-6b350727-a136-448d-ba67-113b53234da7.mp4

Maxr1998 commented:
We already have a loading indicator in the WebViewFragment, so this shouldn't be necessary. Currently, the server state is only `Pending` while the server data is loaded from the database, which should be nearly instantaneous.

susonthapa commented:
I agree, I didn't know about that. I saw a `TODO` in the code and implemented the loader. Since we have implemented the Splash screen I don't think this is necessary so I will close this PR.

Maxr1998 commented:
Yeah, the TODO is quite misleading. I need to ensure if I can just remove it and then do that to prevent any confusion in the future.
